lib/hedwig_mopidy.ex

```diff
@@ -74,4 +74,113 @@ defmodule HedwigMopidy do
+  end
+
+  defmodule Spotify do
+    def get_token do
+      with {:ok, response} <- HTTPoison.post("https://accounts.spotify.com/api/token",
+                                             "grant_type=refresh_token&refresh_token=#{get_refresh_token}",
+                                             ["Authorization": "Basic #{:base64.encode(get_secrets)}",
+                                              "Content-Type": "application/x-www-form-urlencoded"],
+                                             hackney: [pool: :tracklist]) do
+        Poison.decode!(response.body)["access_token"]
+      end
+    end
+
+    def favorite_track(track_uri) do
+      add_track_to_playlist(HedwigMopidy.favorites_playlist, track_uri)
+    end
+
+    def reject_track(playlist_uri, track_uri) do
+      remove_track_from_playlist(playlist_uri, track_uri)
+      add_track_to_playlist(HedwigMopidy.rejects_playlist, track_uri)
+    end
+
+    def add_track_to_playlist(playlist_uri, track_uri) do
+      api_url = transform_playlist_uri(playlist_uri)
+      with {:ok, response} <- HTTPoison.post("#{api_url}?uris=#{track_uri}",
+                                             "",
+                                             ["Authorization": "Bearer #{get_token}",
+                                              "Accept": "application/json"],
+                                             hackney: [pool: :tracklist]) do
+        Poison.decode!(response.body)["snapshot_id"]
+      end
+    end
+
+    def remove_track_from_playlist(playlist_uri, track_uri) do
+      api_url = transform_playlist_uri(playlist_uri)
+      with {:ok, response} <- HTTPoison.request!(:delete,
+                                                 api_url,
+                                                 "{\"tracks\":[{\"uri\":\"#{track_uri}\"}]}",
+                                                 ["Authorization": "Bearer #{get_token}", "Content-Type": "application/json"],
+                                                 hackney: [pool: :tracklist]) do
+        Poison.decode!(response.body)["snapshot_id"]
+      end
+    end
+
+    def get_random_track_uris(playlist_uri, limit \\ 1, track_uris \\ [], count \\ 0) do
+      if count < limit do
+        track_uris = [get_random_track_uri(playlist_uri) | track_uris]
+        get_random_track_uris(playlist_uri, limit, track_uris, count + 1)
+      else
+        track_uris
+      end
+    end
+
+    def get_random_track_uri(playlist_uri) do
+      api_url = transform_playlist_uri(playlist_uri)
+      with {:ok, size_response} <- HTTPoison.get("#{api_url}?limit=1&fields=total",
+                                                 ["Authorization": "Bearer #{get_token}"],
+                                                 hackney: [pool: :tracklist]) do
+        total = Poison.decode!(size_response.body)["total"]
+
+        :random.seed(:os.timestamp)
+        offset = if total > 100 do
```

Comment:
This should probably be a separate function `limit_total` or something, in which case you could call it inline below